### PR TITLE
Add tags package

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vince@vincepri.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/aws/tags/cluster.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package tags
```
Comment (chuckha) [inline pkg/cloud/aws/tags/cluster.go]: this needs  the license and godoc

File: pkg/cloud/aws/tags/tags.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+package tags
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/pkg/errors"
+)
+
+type ApplyParams struct {
```
Comment (chuckha) [inline pkg/cloud/aws/tags/tags.go]: godoc this please

File: pkg/cloud/aws/tags/tags.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		awsTags = append(awsTags, tag)
+	}
+
+	createTagsInput := &ec2.CreateTagsInput{
+		Resources: aws.StringSlice([]string{params.ResourceID}),
+		Tags:      awsTags,
+	}
+
+	_, err := params.EC2Client.CreateTags(createTagsInput)
+	return errors.Wrapf(err, "failed to tag resource %q in cluster %q", params.ResourceID, params.ClusterName)
```
Comment (chuckha) [inline pkg/cloud/aws/tags/tags.go]: didn't realize Wrapf didn't do anything if the error was nil. that's cute.

File: pkg/cloud/aws/tags/tags.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+type BuildParams struct {
```
Comment (chuckha) [inline pkg/cloud/aws/tags/tags.go]: more godoc

File: pkg/cloud/aws/tags/types.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const (
+	// ResourceLifecycleOwned is the value we use when tagging resources to indicate
+	// that the resource is considered owned and managed by the cluster,
+	// and in particular that the lifecycle is tied to the lifecycle of the cluster.
+	ResourceLifecycleOwned = ResourceLifecycle("owned")
+
+	// ResourceLifecycleShared is the value we use when tagging resources to indicate
+	// that the resource is shared between multiple clusters, and should not be destroyed
+	// if the cluster is destroyed.
+	ResourceLifecycleShared = ResourceLifecycle("shared")
```
Comment (chuckha) [inline pkg/cloud/aws/tags/types.go]: is this something we agreed on? or is this something we might need in the future?

Comment (vincepri) [reply pkg/cloud/aws/tags/types.go]: This was in an older PR and was taken from k/k

**Comment by vincepri** [comment]
@chuckha fixed comments, I believe this is good to go.

**Pull request merged**